backend/frontend/static/frontend/utils.js

```diff
@@ -5,3 +5,15 @@ function countLoadedPageItems(parentId) {
   const parent = document.getElementById(parentId);
   return parent.childElementCount - 1; // the button is an element too
 }
+
+/*
+ * Swaps in the body of error pages returned from htmx requests 
+ * for example when an action is forbidden.
+ */
+document.addEventListener("htmx:beforeOnLoad", function (event) {
+  const xhr = event.detail.xhr
+  if (xhr.status == 500 || xhr.status == 403 || xhr.status == 404) {
+    event.stopPropagation() // Tell htmx not to process these requests
+    document.children[0].innerHTML = xhr.response // Swap in body of response instead
```

Comment:
Let's say we have a page that uses htmx to fetch 10 graphs from the backend as it loads. If graph number 8 comes up with an error then the entire page will show that wrror. But the other 9 graphs could be useful.

Generally with frontend dev you want to make sure that errors are well contained. If one little widget is broken you dont want the whole page to break.   Here you seem to be explicitly breaking the whole page. 

What are you trying to achieve with this?

Reply:
Had not thought about this, thank you for pointing it out. I wanted a way to communicate errors for example we have this guard on a view that gets called via ajax:

```python
@user_passes_test_or_forbidden(user_can_request_review)
def action_request_review(request, card_id):
    ...
```

if the request user fails the test, then the user won't know what happened, the error will only be visible in the browser console. 

Is this something we should worry about on this PR or we can leave it out for later?

Reply:
That's a good point. Bt I do think having it fail in this way is much better than breaking hydration. We use hydration already and I'll sure we'll use it more.

I think the thing to do is treat things differently per error. Eg a 500 error is something that really shouldn't happen, it means our backend code fell over. If those happen we can just have an alert saying "something went wrong" or something like that. 

500 errors will get picked up in our error logs and I'll get an email if there are any new ones so things will get fixed.

404s really shouldn't happen in an xhr request because we'll only really be fetching things we are allowed to fetch, with known ids. This would also actually be a bug I think. Same for 403s.  

A 403/4 can also come from a user navigating to a page they are not allowed to see but that should not be handled here.

I think for now it's ok to remove this event handler.